source/adapters/level_zero/queue.cpp

```diff
@@ -504,6 +504,16 @@ UR_APIEXPORT ur_result_t UR_APICALL urQueueCreate(
 
   UR_ASSERT(Context->isValidDevice(Device), UR_RESULT_ERROR_INVALID_DEVICE);
 
+  // optimized path for immediate, in-order command lists
+  if (v2::shouldUseQueueV2() && (Flags & UR_QUEUE_FLAG_SUBMISSION_IMMEDIATE) &&
+      Device->useImmediateCommandLists() &&
+      (Flags & UR_QUEUE_FLAG_OUT_OF_ORDER_EXEC_MODE_ENABLE) == 0) {
+    *Queue = new ur_queue_handle_t_(
+        std::in_place_type<v2::ur_queue_immediate_in_order_t>, Context, Device,
```

Comment:
I have to say, this is an incredibly clunky way of defining a variant of an enum. Can we e.g., create a factory function that would be take a templated variant argument?